config/v1/types_build.go

```diff
@@ -32,17 +32,15 @@ type BuildSpec struct {
 }
 
 type BuildDefaults struct {
-	// GitHTTPProxy is the location of the HTTPProxy for Git source
+	// BuildProxy contains the proxy settings for all build operations, including image pull/push
+	// and source download.
 	// +optional
-	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
+	BuildProxy Proxy `json:"buildProxy,omitempty"`
 
-	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
+	// GitProxy contains the proxy settings for source download operations only. If set, this will override
+	// any BuildProxy settings for source download.
 	// +optional
-	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
-
-	// GitNoProxy is the list of domains for which the proxy should not be used
-	// +optional
-	GitNoProxy string `json:"gitNoProxy,omitempty"`
+	GitProxy Proxy `json:"gitProxy,omitempty"`
```

Comment:
I'm slightly worried this being only git, why not making this more general, I'm not sure if we ever support other VSC-s but why tightly coupling our API to one?

Reply:
Same here for it being a pointer.

Reply:
I'd argue that we _should_ tie these proxies to their underlying protocols:

1. We only support git today - keeping it tied to git makes it clear as to what we support. 
2. It's possible that an admin would need separate proxies per protocol - particularly if those proxy URLs contain auth info. If we only allowed a single agnostic `SourceProxy`, our hands would be tied.

Reply:
I can see both arguments, but also overall I side with @adambkaplan